### PR TITLE
Utils Bindings

### PR DESCRIPTION
**Pull request by codeboy5**
# Pull Request

## Description
I have added a binding for `mean` function. Also added a test in `test_util.py`. @chinmayshah99 is this okay?

## Affected Dependencies
List any dependencies that are required for this change.

## Type of Change
Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation (non-breaking change which adds documentation)
- [x] Improvement (non-breaking change that improves the performance or reliability of existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
`pytest tests`

## Checklist
- [x] I did follow the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md)
- [x] I did follow the [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have added tests for my changes

## Additional Context
Includes bindings for #27

### CODE REVIEW

File: src/bindings/PyDP/algorithms/util.cpp [hunk 1]
```diff
@@ -14,4 +15,6 @@ void init_algorithms_util(py::module& m) {
   util.def("default_epsilon", &dp::DefaultEpsilon);
   util.def("get_next_power_of_two", &dp::GetNextPowerOfTwo);
   util.def("qnorm", &dp::Qnorm);
+  util.def("mean",&dp::Mean<double>);
+  // util.def("hello", &dp:: )
```
Comment (chinmayshah99) [inline src/bindings/PyDP/algorithms/util.cpp]: I think a helper function is needed to covert the python list to a vector. Is this code tested using pytest?

Comment (codeboy5) [reply src/bindings/PyDP/algorithms/util.cpp]: yes I have added a test, you can check that in this commit

Comment (codeboy5) [reply src/bindings/PyDP/algorithms/util.cpp]: @chinmayshah99 should I proceed to change these to have a wrapper function which will allow us to specify the type ?

Comment (chinmayshah99) [reply src/bindings/PyDP/algorithms/util.cpp]: Yeah, please go ahead.

File: src/bindings/PyDP/algorithms/util.cpp [hunk 2]
```diff
@@ -14,4 +15,29 @@ void init_algorithms_util(py::module& m) {
...
+  util.def("mean", &dp::Mean<int>);
+  util.def("variance", &dp::Variance<double>);
+  util.def("standard_deviation", &dp::StandardDev<double>);
+  util.def("order_statistics", &dp::OrderStatistic<double>);
+  util.def("correlation", &dp::Correlation<double>);
+  util.def("vector_filter", &dp::VectorFilter<double>);
+  util.def("vector_to_string", &dp::VectorToString<double>);
+  util.def("round_to_nearest_multiple", &dp::RoundToNearestMultiple);
+  // TODO: Throw an exception in case the function returns false.
+  util.def("safe_add", [](int64_t i, int64_t j) {
```
Comment (chinmayshah99) [inline src/bindings/PyDP/algorithms/util.cpp:27]: How to change datatype when we want to use floats?

Comment (codeboy5) [reply src/bindings/PyDP/algorithms/util.cpp:27]: If we do something like `(double i, double j){}` it would automatically use that if passed a float.

Comment (chinmayshah99) [reply src/bindings/PyDP/algorithms/util.cpp:27]: So the way we are supposed to use this is: 
`safe_add(number1, number2, 'int'/'float')`

Comment (codeboy5) [reply src/bindings/PyDP/algorithms/util.cpp:27]: How would we do something like this in plain c++ ?

**Comment by codeboy5** [comment]
> # Pull Request
> ## Description
> I have added a binding for `mean` function. Also added a test in `test_util.py`. @chinmayshah99 is this okay?
> 
> ## Affected Dependencies
> List any dependencies that are required for this change.
> 
> ## Type of Change
> Please mark options that are relevant.
> 
> * [ ]  Bug fix (non-breaking change which fixes an issue)
> * [ ]  Documentation (non-breaking change which adds documentation)
> * [x]  Improvement (non-breaking change that improves the performance or reliability of existing functionality)
> * [x]  New feature (non-breaking change which adds functionality)
> * [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
> 
> ## How has this been tested?
> `pytest tests`
> 
> ## Checklist
> * [x]  I did follow the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md)
> * [x]  I did follow the [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
> * [ ]  I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
> * [x]  I have added tests for my changes
> 
> ## Additional Context
> Includes bindings for #27

@chinmayshah99 I have added exceptions for the invalid operations.

File: src/bindings/PyDP/algorithms/util.cpp [hunk 3]
```diff
@@ -14,4 +15,32 @@ void init_algorithms_util(py::module& m) {
...
+  util.def("mean", &dp::Mean<double>);
+  util.def("mean", &dp::Mean<int>);
+  util.def("variance", &dp::Variance<double>);
+  util.def("standard_deviation", &dp::StandardDev<double>);
+  util.def("order_statistics", &dp::OrderStatistic<double>);
+  util.def("correlation", &dp::Correlation<double>);
+  util.def("vector_filter", &dp::VectorFilter<double>);
+  util.def("vector_to_string", &dp::VectorToString<double>);
+  util.def("round_to_nearest_multiple", &dp::RoundToNearestMultiple);
+  // TODO: Throw an exception in case the function returns false.
```
Comment (chinmayshah99) [inline src/bindings/PyDP/algorithms/util.cpp]: Is this still a TODO?

**Review by chinmayshah99 — DISMISSED** [review]
LGTM!

**Comment by chinmayshah99** [comment]
@codeboy5 if it's no longer WIP, can we remove the tag from the PR title?

**Comment by codeboy5** [comment]
> @codeboy5 if it's no longer WIP, can we remove the tag from the PR title?

Yes I think this is done from my side!

File: src/bindings/PyDP/algorithms/util.cpp (hunk 3 above)
Comment (chinmayshah99) [reply src/bindings/PyDP/algorithms/util.cpp]: @codeboy5 can you remove this comment if it's done already or is it left as a future exercise?

**Review by chinmayshah99 — APPROVED** [review]

**Pull request merged**